extensions/amp-story/1.0/amp-story-reaction-quiz.js

```diff
@@ -0,0 +1,221 @@
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+      this.rootEl_.removeChild(promptContainer);
+    } else {
+      const prompt = document.createElement(promptInput.tagName);
+
+      prompt.textContent = promptInput.textContent;
+      prompt.classList.add('i-amphtml-story-reaction-quiz-prompt');
+
+      this.element.removeChild(promptInput);
+      promptContainer.appendChild(prompt);
+    }
+
+    const options = toArray(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      dev().error(TAG, 'Improper number of options');
+    }
+
+    // Localize the answer choice options
+    this.answerChoiceOptions_ = this.answerChoiceOptions_.map((choice) => {
+      return this.localizationService_.getLocalizedString(
+        LocalizedStringId[`AMP_STORY_QUIZ_ANSWER_CHOICE_${choice}`]
+      );
+    });
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      dev().error(TAG, 'Too many children');
+    }
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the quiz element.
+   *
+   * @param {Element} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const convertedOption = buildOptionTemplate(dev().assertElement(option));
+
+    // Fill in the answer choice and set the option ID
+    convertedOption.querySelector(
+      '.i-amphtml-story-reaction-quiz-answer-choice'
+    ).textContent = this.answerChoiceOptions_[index];
+    convertedOption.optionIndex_ = index;
+
+    // Extract and structure the option information
+    const optionText = document.createElement('span');
+    optionText.classList.add('i-amphtml-story-reaction-quiz-option-text');
+    optionText.textContent = option.textContent;
+    convertedOption.appendChild(optionText);
+
+    // Add text container for percentage display
+    const percentageText = document.createElement('span');
+    percentageText.classList.add(
+      'i-amphtml-story-reaction-quiz-percentage-text'
+    );
+    convertedOption.appendChild(percentageText);
+
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    this.element.removeChild(option);
+
+    this.rootEl_
+      .querySelector('.i-amphtml-story-reaction-quiz-option-container')
+      .appendChild(convertedOption);
+  }
+
+  /**
+   * Get the quiz element
+   * @return {Element}
+   */
+  getQuizElement() {
+    return this.rootEl_;
+  }
+
+  /**
+   * @override
+   */
+  updateOptionPercentages_() {
+    if (!this.responseData_) {
+      return;
+    }
+
+    const options = toArray(
+      this.rootEl_.querySelectorAll('.i-amphtml-story-reaction-quiz-option')
+    );
+
+    const percentages = this.preprocessPercentages_(this.responseData_);
+
+    this.responseData_['responses'].forEach((response) => {
```

Comment:
Will need to refactor the backend connection and calculations based on assumptions and API decisions in design doc, in future PR